src/Application.php

```diff
@@ -59,9 +57,17 @@ public function addInitRoboFileCommand($roboFile, $roboClass)
         $this->add($createRoboFile);
     }
 
-    protected function addSelfUpdateCommand()
+    /**
+     * Add self update command, do nothing if null is provided
+     *
+     * @param string $repository GitHub Repository for self update
+     */
+    public function addSelfUpdateCommand($repository = null)
     {
-        $selfUpdateCommand = new SelfUpdateCommand('self:update', Robo::VERSION, 'consolidation/robo');
+        if ($repository == null) {
+            return;
+        }
+        $selfUpdateCommand = new SelfUpdateCommand('self:update', Robo::VERSION, $repository);
```

Comment:
Use `$version` from the constructor here. (You'll probably need to cache it in a class field.) Also inject the app name from the constructor here.